README.md

```diff
@@ -1,2 +1,64 @@
-# harry-potter-quarkus
-Showcase in a web application Quarkus and Infinispan
+# Quarkus demo: Infinispan Client
+
+This example showcases how to use Infinispan client with Quarkus.
+* Create stores dynamically
+* Simple get/put operations
+* Full-text query
+* Continuous query
+
+Uses quarkus features, such as
+* REST web-service
+* Web Socket
+* Task Scheduling
+* Application properties
+* Application lifecycle events listeners
+
+# Run infinispan server
+
+- Running with docker `docker run -it -p 11222:11222 jboss/infinispan-server:latest`
+- Download the server from `http://www.infinispan.org` and run `./bin/standalone.sh`
+
+# Harry-Potter demo
+
+The demo will load characters and spells. Characters can execute spells, and we want to monitor who is in
+Hogwarts and performing magic.
+
+A Character has an id, name, biography and type (other, student, teacher, muggle).
+A Spell has an id, name and description.
+A Spell is performed by a character. This character may or may not be at Hogwarts.
+
+## Data loading
+`DataLoader` class loads characters and spells in two separate stores.
+This is done on startup time
+
+## Magic is going on
+[Hogwarts Magic Creator](src/main/java/org/acme/infinispanclient/service/HogwartsMagicCreator.java) is going to emulate
+characters performing some magic. It will randomly pick a character and a spell to perform (if they can)!
+Characters in Hogwarts are teachers or students. 
+
+## Search
+[A simple REST service](src/main/java/org/acme/infinispanclient/CharactersResource.java) is available to query
+characters by id or perform a full-text search on top of the name or the biography. 
+
+## Magic Socket
+[A socket](src/main/java/org/acme/infinispanclient/HogwartsMagicSocket.java) that performs a Continuous Query making it possible to
+monitor which characters in Hogwarts are currently performing magic.
+
+# Run the demo in dev mode
+To run in development mode, just run `mvn clean compile quarkus:dev`. This allows for hot swapping the application, so you can tweak parts of it and it will automatically redeploy.
+Go to `http://localhost:8081` and monitor the magic!
+You will be connected to the monitoring socket and links to perform a REST based search.
```

Comment:
There's something weird with this sentence, probably a verb missing or something: "be connected to... and links"

Reply:
yes, sounds weird, reformulating, thanks!